src/util/transition/ngbCollapseTransition.ts

```diff
@@ -48,30 +49,40 @@ export const ngbCollapsingTransition: NgbTransitionStartFn<NgbCollapseCtx> =
         return;
       }
 
-      // No maxHeight -> running the transition for the first time
-      if (!maxHeight) {
-        maxHeight = measureCollapsingElementHeightPx(element);
-        context.maxHeight = maxHeight;
-
-        // Fix the height before starting the animation
-        element.style.height = direction !== 'show' ? maxHeight : '0px';
-
+      if (direction === 'show') {
+        // Fix the dimension before starting the animation
+        element.style[dimension] = '0px';
         classList.remove('collapse');
-        classList.remove('collapsing');
-        classList.remove('show');
-
-        reflow(element);
 
         // Start the animation
         classList.add('collapsing');
-      }
 
-      // Start or revert the animation
-      element.style.height = direction === 'show' ? maxHeight : '0px';
+        const scrollDimension = `scroll${dimension[0].toUpperCase()}${dimension.slice(1)}`;
```

Comment:
Fell free to ignore this. Personally I would find it simpler/more readable to do something like this:

```ts
const SCROLL_DIMENSIONS = {
  'height': 'scrollHeight',
  'width': 'scrollWidth'
};

element.style[dimension] = element[SCROLL_DIMENSIONS[dimension]] + 'px';
```

Reply:
I would agree with you, but I wanted to remain consistent with the Bootstrap code as much as possible, that's why I went for this approach